Helper/AdapterHelper.php

```diff
@@ -298,16 +104,16 @@ public function isAllowed()
         $storeId = $this->getStoreId();
 
         return
-            $this->config->getApplicationID($storeId)
-            && $this->config->getAPIKey($storeId)
-            && $this->config->isEnabledFrontEnd($storeId)
-            && $this->config->makeSeoRequest($storeId);
+            $this->algoliaHelper->getConfigHelper()->getApplicationID($storeId)
```

Comment:
Provision of `configHelper` shouldn't be a responsibility of `algoliaHelper`. `AlgoliaHelper` should handle only Algolia-related stuff, so I don't like that it provisions `configHelper` here. The `__constructor` has 3 arguments now, which is great, but we need to think about readability and what helpers are responsible for what.

I'd much rather inject `ConfigHelper` here or via `Magento\CatalogSearch\Helper\Data` class, which makes more sense as the `CatalogSearch helper` is the class responsible for the correct search in general. WDYT?